testplan/common/utils/networking.py

```diff
@@ -8,10 +8,65 @@
     absolute_import,
 )
 import socket
+import ipaddress
 
 
 def port_to_int(port):
+    """Convert a port string to an integer."""
     try:
         return int(port)
     except ValueError:
         return socket.getservbyname(port)
+
+
+def format_access_urls(host, port, url_path):
+    """
+    Format a string to log to give HTTP access to a URL endpoint listening
+    on a particular host/port. Handles special 0.0.0.0 IP and formats IPV6
+    addresses. The returned string is indented by 4 spaces.
+
+    :param host: hostname or IP address
+    :type host: ``str``
+    :param port: port number
+    :type port: ``int``
+    :param url_path: path to format after host:port part of URL (expected to
+        contain leading "/" e.g "/index.html")
+    :type url_path: ``str``
+    :return: a formatted string containing the connection URL(s)
+    :rtype: ``str``
+    """
+    # Handle 0.0.0.0 as a special case: this means that the URL can be accessed
+    # both via localhost or on any IP address this host owns.
```

Comment:
Shouldn't this be "this means that the URL can be accessed from any IP address"? As long as a client is able to access the server's listening port (i.e. not blocked by firewall) 0.0.0.0 means any subnet can connect.

Reply:
0.0.0.0 in this context means "all local addresses on the machine". So if a server has multiple physical ethernet links, each with their own IP address, then the HTTP request can be sent to any of those IP addresses. It's not about the IP address of the client.

Reply:
Right, in context makes sense.